apache/superset
# refactor(bulk_select): Fix bulk select tagging issues for users

### fix(bulk_select): allow users to tag a saved query if using bulk select

### SUMMARY
This PR addresses an issue where  users were unable to tag objects despite having the necessary permissions. The changes ensure that the tagging functionality correctly validates user ownership or edit permissions and allows tagging where applicable. Additionally, it refines error handling to avoid unnecessary warnings while maintaining robust validation for invalid objects.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
![1736432737670 (1)](https://github.com/user-attachments/assets/2059d533-9370-43da-b65b-626c38da61b9)





### TESTING INSTRUCTIONS
1. Create a saved query
2. Go to saved query tab
3. Click bulk select
4. Select the created query
5. Click Add Tag
6. Add any tag
Note that you'll need to have the feature flag TAGGING_SYSTEM enabled and create a tag in the Tags page (in Settings) first before going through the repro steps.

### ADDITIONAL INFORMATION


- [x] Has associated issue: Fixes #27988
- [ ] Required feature flags:The **TAGGING_SYSTEM** flag must be enabled.
